File: backend/ModuMessenger/src/main/java/com/example/modumessenger/chat/dto/ChatType.java
package com.example.modumessenger.chat.dto;


import java.util.HashMap;
import java.util.Map;

public enum ChatType {
    TEXT(1, ""),
    IMAGE(2, "image"),
    FILE(3, "file"),
    AUDIO(4, "audio");

    private final int chatType;
    private final String chatTypeStr;

    private static final Map<Integer, ChatType> chatTypeMap = new HashMap<>();

    static {
        for(ChatType chatType : values()) {
            chatTypeMap.put(chatType.getChatType(), chatType);
        }
    }

    ChatType(int chatType, String chatTypeStr) {
        this.chatType = chatType;
        this.chatTypeStr = chatTypeStr;
    }

    public int getChatType() { return this.chatType; }
    public String getChatTypeStr() { return this.chatTypeStr; }


    public static ChatType fromChatType(int type) {
        return chatTypeMap.get(type);
    }
}

File: backend/chat-service/src/main/kotlin/com/example/chatservice/chat/dto/ChatType.kt
package com.example.chatservice.chat.dto

enum class ChatType(val chatType: Int, val chatTypeStr: String) {
    TEXT(1, ""),
    IMAGE(2, "image"),
    FILE(3, "file"),
    AUDIO(4, "audio"),
    ;

    companion object {
        private val chatTypeMap: Map<Int, ChatType> = entries.associateBy { it.chatType }

        @JvmStatic
        fun fromChatType(type: Int): ChatType? = chatTypeMap[type]
    }
}
